src/PrestaShopBundle/Form/Admin/Improve/Payment/Preferences/PaymentModuleCountryRestrictionsType.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Improve\Payment\Preferences;
+
+use PrestaShop\PrestaShop\Adapter\Country\CountryDataProvider;
+use PrestaShopBundle\Form\Admin\Type\Material\MaterialMultipleChoiceTableType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class PaymentModuleCountryRestrictionsType defines Country Restriction form in "Improve > Payment > Preferences" page.
+ */
+class PaymentModuleCountryRestrictionsType extends PaymentModuleRestrictionsParentType
+{
+    /**
+     * @var array
+     */
+    private $countryChoices;
+
+    /**
+     * @var CountryDataProvider
+     */
+    private $countryDataProvider;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param array $paymentModules
```

Comment:
We need to define clearly what this array contains:
```suggestion
     * @param array<string, ModuleInterface> $paymentModules
```

Reply:
And this type should also be improved in:
- `PaymentModuleRestrictionsParentType`
- `PaymentModuleListProviderInterface`
- any form that extends `PaymentModuleRestrictionsParentType`